contracts/StorageAccessible.sol

```diff
@@ -52,20 +52,47 @@ contract StorageAccessible {
      */
     function simulateDelegatecall(
         address targetContract,
-        bytes memory calldataPayload
+        bytes calldata calldataPayload
     ) public returns (bytes memory response) {
```

Comment:
Could we make this `external` and then just use the `return` opcode. This way we would not have to care about memory management and we would also not need put the length into the returndata in https://github.com/gnosis/util-contracts/pull/48/files#diff-89f9ef055c55ab6ad50077b9f0119fdeb5c302d94d905ebaa3aca0dfc16015dcR122

Reply:
This is a nice idea, and I actually started implementing it the way you described. The reason I shifted gears was that I intend on moving this method to a `library`, where it would be an `internal` method and would require the logic to allocate a buffer for the response.

I'm fine with changing it to your suggestion and then changing it back to this once I move this to a library.

Reply:
Fine either way for me, I assume the savings are minimal :)

Reply:
Is that :+1: for changing it and then later changing it back or leaving it as is?

Reply:
the 👍 was for "I understand what you mean". As said I don't have a strong opinion. I would probably avoid unnessary back and forth and keep it as is, so that it is easier to later extract it into a library.

Reply:
Cool less work for me :sweat_smile: